gradio/interface.py

```diff
@@ -447,10 +454,82 @@ def clean_html(raw_html):
         if self.analytics_enabled:
             utils.initiated_analytics(data)
 
-        # Alert user if a more recent version of the library exists
         utils.version_check()
         Interface.instances.add(self)
 
+        param_names = inspect.getfullargspec(self.predict[0])[0]
+        for component, param_name in zip(self.input_components, param_names):
+            if component.label is None:
+                component.label = param_name
+        for i, component in enumerate(self.output_components):
+            if component.label is None:
+                component.label = "output_" + str(i)
+
+        if self.allow_flagging != "never":
+            self.flagging_callback.setup(
+                self.input_components + self.output_components, self.flagging_dir
+            )
+
+        self.blocks = Blocks()
+        with self.blocks:
+            if self.title:
+                Markdown(
+                    "<h1 style='text-align: center; margin-bottom: 1rem'>"
+                    + self.title
+                    + "</h1>"
+                )
+            if self.description:
+                Markdown(self.description)
+            with Row():
+                with Column(
+                    css={
+                        "background-color": "rgb(249,250,251)",
+                        "padding": "0.5rem",
+                        "border-radius": "0.5rem",
+                    }
+                ):
+                    for component in self.input_components:
+                        Block.__init__(component)
+                    with Row():
+                        submit_btn = Button("Submit")
+                        clear_btn = Button("Clear")
+                with Column(
+                    css={
+                        "background-color": "rgb(249,250,251)",
+                        "padding": "0.5rem",
+                        "border-radius": "0.5rem",
+                    }
+                ):
+                    for component in self.output_components:
+                        Block.__init__(component)
+                    with Row():
+                        flag_btn = Button("Flag")
+            if self.examples:
+                examples = DatasetViewer(
+                    components=self.input_components, samples=self.examples
+                )
+            submit_btn.click(
+                lambda *args: self.process(args)[0][0]
+                if len(self.output_components) == 1
+                else self.process(args)[0],
+                self.input_components,
+                self.output_components,
+            )
+            clear_btn.click(
+                lambda: [None]
+                * (len(self.input_components) + len(self.output_components)),
+                [],
+                self.input_components + self.output_components,
+            )
+            examples.click(
```

Comment:
should only happen if `examples` is not `None`. Otherwise, it will throw an error.